### PR TITLE
pin/unpin directly in columns header

### PR DESCRIPTION
**Pull request by jserve**
I got asked for our Mastodon instance to introduce the pin/unpin being directly in the header of the columns, so I added this option to our instance and thought that it's a good idea to share it, perhaps someone else would like it too.

The user can select at his own wether he want to have the pin/unpin in the column header, right now it's default to true.

### CODE REVIEW

**Comment by nightpool** [comment]
Hey, just a heads up that if you want to make changes to a pull request, you should push new commits to the branch it's based on, not close it and open a new one. That makes it harder to review and keep track of

File: app/javascript/mastodon/components/column_header.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { FormattedMessage, injectIntl, defineMessages } from 'react-intl';
+import {connect} from 'react-redux';
```
Comment (nightpool) [inline app/javascript/mastodon/components/column_header.js]: should be `{ connect }` according to eslint

File: app/javascript/mastodon/components/column_header.js [hunk 2]
```diff
@@ -10,6 +11,15 @@ const messages = defineMessages({
   moveRight: { id: 'column_header.moveRight_settings', defaultMessage: 'Move column to the right' },
 });
 
+const makeMapStateToProps = () => {
+    const mapStateToProps = state => ({
```
Comment (nightpool) [inline app/javascript/mastodon/components/column_header.js]: two space indent

File: app/javascript/mastodon/components/column_header.js [hunk 3]
```diff
@@ -10,6 +11,15 @@ const messages = defineMessages({
...
+	systemHdrPin: state.getIn(['meta', 'system_hdr_pin']),
```
Comment (nightpool) [inline app/javascript/mastodon/components/column_header.js]: needs indentation here.

File: app/javascript/mastodon/components/column_header.js [hunk 4]
```diff
@@ -101,6 +113,9 @@ export default class ColumnHeader extends React.PureComponent {
     if (multiColumn && pinned) {
       pinButton = <button key='pin-button' className='text-btn column-header__setting-btn' onClick={onPin}><i className='fa fa fa-times' /> <FormattedMessage id='column_header.unpin' defaultMessage='Unpin' /></button>;
 
+      if(systemHdrPin)
+          pinButtonHdr = <button key='pin-button' className='column-header__button' onClick={onPin}><i className='fa fa fa-times' /></button>;
```
Comment (nightpool) [inline app/javascript/mastodon/components/column_header.js]: please use brackets around your if statements.

Comment (jserve) [reply app/javascript/mastodon/components/column_header.js]: Thanks for the assisting, I will adjust my files and will turn in a new PR

**Comment by jserve** [comment]
I will close for now the PR, I will come back when I am more familar with all the git stuff :)

**Pull request closed without merging**